File: src/indi/indi_del_property.c
/* NyxNode
 * Author: Jérôme ODIER <jerome.odier@lpsc.in2p3.fr>
 * SPDX-License-Identifier: GPL-2.0-only (Mongoose backend) or GPL-3.0+
 */

/*--------------------------------------------------------------------------------------------------------------------*/

#include "../nyx_node_internal.h"

/*--------------------------------------------------------------------------------------------------------------------*/

nyx_dict_t *nyx_del_property_new(
    STR_t device,
    STR_t name,
    STR_t message
) {
    nyx_dict_t *result = nyx_dict_new();

    /*----------------------------------------------------------------------------------------------------------------*/

    char timestamp[32];

    internal_get_timestamp(sizeof(timestamp), timestamp);

    /*----------------------------------------------------------------------------------------------------------------*/

    nyx_dict_set_string_managed_unref(result, "<>", nyx_string_dup("delProperty"));

    nyx_dict_set_string_managed_unref(result, "@client", nyx_string_dup("unknown"));
    nyx_dict_set_string_managed_unref(result, "@device", nyx_string_dup(device));
    nyx_dict_set_string_managed_unref(result, "@timestamp", nyx_string_dup(timestamp));

    /*----------------------------------------------------------------------------------------------------------------*/

    if(name != NULL) {
        nyx_dict_set_string_managed_unref(result, "@name", nyx_string_dup(name));
    }

    if(message != NULL) {
        nyx_dict_set_string_managed_unref(result, "@message", nyx_string_dup(message));
    }

    /*----------------------------------------------------------------------------------------------------------------*/

    return result;
}

/*--------------------------------------------------------------------------------------------------------------------*/

File: nyx_node/indi/indi_del_property.py
# -*- coding: utf-8 -*-
########################################################################################################################
# NyxNode
# Author: Jérôme ODIER <jerome.odier@lpsc.in2p3.fr>
# SPDX-License-Identifier: GPL-3.0+
########################################################################################################################

from .. import bind
from .. import json

from . import utils

########################################################################################################################

@utils.nyx_property(
    'device',
    '@device',
)
@utils.nyx_property(
    'name',
    '@name',
)
@utils.nyx_property(
    'message',
    '@message',
)
class NyxDelProperty(json.json_dict.NyxDict):

    ####################################################################################################################

    def __init__(self, device: str, name: str | None = None, message: str | None = None):

        super().__init__(bind.lib.nyx_del_property_new(
            bind.as_bytes(device, allow_none = False),
            bind.as_bytes(name),
            bind.as_bytes(message),
        ))

########################################################################################################################

__all__ = [name for name in globals() if name.lower().startswith('nyx')]

########################################################################################################################
